packages/gatsby/src/utils/webpack.config.js

```diff
@@ -326,7 +326,7 @@ module.exports = async (
       modulesDirectories: [
         directoryPath(`node_modules`),
         `node_modules`,
-        directoryPath(`node_modules`, `gatsby`, `node_modules`),
+        path.join(__dirname, `..`, `..`, `node_modules`),
```

Comment:
this also shouldn't look in the gatsby sites `node_modules`, gatsby may not be there (could be hoisted).